pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseBrokerRequestHandler.java

```diff
@@ -121,82 +122,89 @@ public BaseBrokerRequestHandler(PinotConfiguration config, String brokerId, Brok
   public BrokerResponse handleRequest(JsonNode request, @Nullable SqlNodeAndOptions sqlNodeAndOptions,
       @Nullable RequesterIdentity requesterIdentity, RequestContext requestContext, @Nullable HttpHeaders httpHeaders)
       throws Exception {
-    requestContext.setBrokerId(_brokerId);
-    long requestId = _requestIdGenerator.get();
-    requestContext.setRequestId(requestId);
+    try (QueryThreadContext.CloseableContext closeMe = QueryThreadContext.open()) {
```

Comment:
Given that I had to introduce a try-with-resources here, I recommend to review this class with the hide whitespace option in GH